core/src/views/radio_buttons.rs

```diff
@@ -0,0 +1,118 @@
+use crate::{Context, Entity, Event, Handle, View, WindowEvent, MouseButton, PseudoClass, Label, LayoutType};
+use crate::style::PropSet;
+use crate::Units::*;
+
+pub struct RadioButtons {
+    on_changed: Option<Box<dyn Fn(&mut Context, usize)>>,
+}
+
+const ICON_CHECK: &str = "\u{00B7}";
+
+impl RadioButtons {
+    pub fn with_icons<F>(
+        cx: &mut Context,
+        idx: usize,
+        idx_count: usize,
+        builder: F,
+        icon_checked: String,
+        icon_unchecked: String,
+    ) -> Handle<Self>
+    where
+        F: 'static + Fn(&mut Context, usize) + Clone
+    {
+        Self {
+            on_changed: None,
+        }
+            .build2(cx, move |cx| {
+                for idx_available in 0..idx_count {
```

Comment:
There's a subtle quirk when you create a variable number of sub-widgets inside of the build closure. Essentially vizia uses the tree to determine if a widget should be created or just updated. Because this widget could be within a binding which changes the `idx_count`, and thus could change the number of `RadioButton` widgets created, it will cause a bug because the tree won't match. 

The easiest solution to this is to remove the children of the widget before creating any new ones, so essentially forcing it to rebuild its children rather than update them. For this reason using a `for` loop or an `if` statement to selectively build widgets doesn't work in the general case. For the for loop you can use the `ForEach` view, which does the child removal part for you.